### PR TITLE
TAUR-797 rewrite create numenta rpm

### PR DESCRIPTION
**Pull request by shantanoo**
Cc: @unixorn

### CODE REVIEW

File: infrastructure/infrastructure/utilities/numenta_rpm.py [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+from infrastructure.utilities.exceptions import InvalidParametersError
+from infrastructure.utilities.path import (
+  changeToWorkingDir,
+  purgeDirectory,
+  rmrf)
+from infrastructure.utilities.cli import runWithOutput
+
+
+class NumentaRPM(object):
+  """docstring for NumentaRPM
```
Comment (unixorn) [inline infrastructure/infrastructure/utilities/numenta_rpm.py]: Make the docstring more specific, please.

**Comment by unixorn** [comment]
Once the docstring gets updated, lgtm and you can merge.

**Comment by shantanoo** [comment]
@unixorn added support for passing `dict` object during class initialization. `config` members are being accessed by `dot` notation (e.g. `config.varName`). During initialization, `dict` will be converted to `class` object. Keys are variables of the `class`. Please review.

**Comment by shantanoo** [comment]
Cc: @unixorn

File: infrastructure/infrastructure/utilities/numenta_rpm.py [hunk 2]
```diff
@@ -0,0 +1,546 @@
+import os
```
Comment (jcasner) [inline infrastructure/infrastructure/utilities/numenta_rpm.py]: Need to add the standard copyright header

File: infrastructure/infrastructure/utilities/numenta_rpm.py [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+
+from infrastructure.utilities import git
+from infrastructure.utilities import logger as log
+from infrastructure.utilities import rpm
+from infrastructure.utilities.exceptions import InvalidParametersError
+from infrastructure.utilities.path import (
+  changeToWorkingDir,
+  purgeDirectory,
+  rmrf)
+from infrastructure.utilities.cli import runWithOutput
```
Comment (jcasner) [inline infrastructure/infrastructure/utilities/numenta_rpm.py]: alphabetize plz

File: infrastructure/infrastructure/utilities/numenta_rpm.py [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+    if failmsg:
+      raise InvalidParametersError(failmsg)
+    self.config = config
+    self.environment = dict(os.environ)
+    self.fakeroot = None
+    self.logger = log.initPipelineLogger(name="create-numenta-rpm",
+                                         logLevel=config.logLevel)
+    self.productsDirectory = None
+
+  def cleanupDirectories(self):
```
Comment (jcasner) [inline infrastructure/infrastructure/utilities/numenta_rpm.py]: 2 newlines between functions within a class

File: infrastructure/infrastructure/utilities/numenta_rpm.py [hunk 5]
```diff
@@ -0,0 +1,546 @@
...
+    Numenta convention is to only include explicitly whitelisted formulas
+    and files in RPMs deployed to customer machines.
+
+    We add a PUBLIC file at the top level of a formula's directory tree
+    to add it to the whitelist.
+
+    This prevents us from accidentally publishing internal-only files to
+    customer machines.
+
+    @param saltpath - Path to /srv/salt in the fakeroot
```
Comment (jcasner) [inline infrastructure/infrastructure/utilities/numenta_rpm.py]: The Numenta format is `:param <param_name>: definition`

File: infrastructure/infrastructure/utilities/numenta_rpm.py [hunk 6]
```diff
@@ -0,0 +1,546 @@
...
+    findPemFiles = """find %s -name *.pem -exec rm -fv {} ;""" % saltpath
+    logger.debug("**************************************************")
+    logger.debug("Sanitizing %s with %s", saltpath, findPubkeys)
+    runWithOutput(findPemFiles, logger=logger)
+
+  def constructFakeroot(self):
+    """
+    Construct a fakeroot.
+
+    @returns (iteration, fakerootSHA) where iteration is the total commit count
```
Comment (jcasner) [inline infrastructure/infrastructure/utilities/numenta_rpm.py]: Please use the correct formatting for `:returns:` and `:rtype:`

File: infrastructure/infrastructure/utilities/numenta_rpm.py [hunk 7]
```diff
@@ -0,0 +1,546 @@
...
+    if flavor == "prebuiltgrok":
+      return self.constructPreBuiltGrokFakeroot()
+
+
+  def constructPreBuiltGrokFakeroot(self):
+    """
+      construct fakeroot from prebuilt grok
+
+      @returns SHA of the products repo in the fakeroot
+
```
Comment (jcasner) [inline infrastructure/infrastructure/utilities/numenta_rpm.py]: No need for the new line here

Comment (jcasner) [inline infrastructure/infrastructure/utilities/numenta_rpm.py]: Add `:rtype:`

**Comment by jcasner** [comment]
:+1:

**Pull request merged**